File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        
        int n = nums.size();
        
        int lmax=0,gmax=0;
        
        for(int i=0;i<n;i++)
        {
            if(nums[i]==1)
            {
                lmax++;
            }
            else
            {
                gmax= max(gmax,lmax);
                lmax = 0;
            }
        }
        
        return max(gmax,lmax);
        
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int maxi=0,sum=0;
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i]==0) sum=0;
            else
            {
                sum+=nums[i];
                maxi=Math.max(sum,maxi);
            }
        }
        return maxi;
        
    }
}